src/main/java/com/gmail/chernii/oleksii/flyweight/ShapeFactory.java

```diff
@@ -0,0 +1,25 @@
+package com.gmail.chernii.oleksii.flyweight;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Created by Space on 12.04.2019.
+ */
+public class ShapeFactory {
+    private static final Map<String, Shape> shapes = new HashMap<>();
+
+    public Shape getShape(String shapeName) {
+        Shape shape = shapes.get(shapeName);
+        if (shape == null) {
+            if (shapeName.equalsIgnoreCase("point")) {
```

Comment:
better use constants fot these purpose
and write like this `POINT.equalsIgnoreCase(shapeName)` because if shapeName is null you will get NPE